src/drive/ducks/services/components/SuggestionProvider.jsx

```diff
@@ -77,4 +79,34 @@ class SuggestionProvider extends React.Component {
   }
 }
 
+function getIconUrl(mimetype) {
+  const iconsContext = require.context(
+    '../../../assets/icons/',
+    false,
+    /icon-type-.*.svg$/
+  )
+  const icons = iconsContext.keys().reduce(
+    (acc, item) => ({
+      ...acc,
+      [item.replace(/\.\/icon-type-(.*)\.svg/, '$1')]: iconsContext(item)
+    }),
+    {}
+  )
+
+  const mapMimeIcon = {
+    'image/jpeg': 'image',
+    'image/png': 'image',
+    'application/pdf': 'pdf'
+  }
+
```

Comment:
`mapMimeIcon`, the `iconsContext` and `icons` variables should be outside the function otherwise you build them every time you need an icon.

Reply:
Crap, I wanted to isolate the code to be easily movable.